### ISSUE TITLE
custom_projection_example.py is completely out of date

### ISSUE THREAD

**Issue by anntzer** [matplotlib/matplotlib#6865]
Basically the whole example (http://matplotlib.org/examples/api/custom_projection_example.html) seems to be an outdated implementation of the Hammer projection (available at https://github.com/matplotlib/matplotlib/blob/master/lib/matplotlib/projections/geo.py).  In fact it is so outdated that it doesn't manage to make the lines become "curved" like they should.  Instead, just copy-pasting the `if __name__ == "__main__"` block at the end to use the standard `hammer` projection results in a correct plot:
```
import matplotlib.pyplot as plt

plt.subplot(111, projection="hammer")
p = plt.plot([-1, 1, 1], [-1, -1, 1], "o-")
plt.grid(True)

plt.show()
```
![hammer](https://cloud.githubusercontent.com/assets/1322974/17275274/c6941644-56b6-11e6-8929-9b3d854eed54.png)

**Comment by tacaswell** [comment]
This is broken back as far as we have hosted built docs: http://matplotlib.org/1.2.1/examples/api/custom_projection_example.html

:+1: on reducing the example to the code in the OP of this issue.

**Comment by tacaswell** [comment]
Looks like this is mostly @mdboom 's work from 2008.

**Comment by NelleV** [comment]
If I am not mistaken, this is fixed in master. @anntzer care to check?

**Comment by QuLogic** [comment]
Fixed by #7111.

**Comment by anntzer** [comment]
I'm not convinced that it was worth copying the entire implementation of the Hammer projection into the example but I guess it's fine too.
